deltacat/utils/profiling.py

```diff
@@ -0,0 +1,39 @@
+import ray
+from dataclasses import dataclass
+
+
+@dataclass
+class ClusterResourcesUsage:
+    def __init__(
+        self,
+        total_memory_bytes: float,
+        used_memory_bytes: float,
+        total_cpu: float,
+        used_cpu: float,
+        total_object_store_memory_bytes: float,
+        used_object_store_memory_bytes: float,
+    ):
+        self.total_memory_bytes = total_memory_bytes
+        self.used_memory_bytes = used_memory_bytes
+        self.total_cpu = total_cpu
+        self.used_cpu = used_cpu
+        self.total_object_store_memory_bytes = total_object_store_memory_bytes
+        self.used_object_store_memory_bytes = used_object_store_memory_bytes
+
+
+def get_current_cluster_resources_usage():
```

Comment:
```suggestion
def get_cluster_utilization():
```

Same comment as above for class naming. Also, if the primary purpose of this function is to construct the corresponding class, then should we just make it a static factory method attached to the `ClusterUtilization` class?

Reply:
Yeah, that makes sense.